File: Revision/STL/stack.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    // basic operation
    stack<int> s;

    // push and pop
    s.push(1);
    s.push(2);
    s.push(3);
    s.push(4);

    s.pop();
    s.pop();

    cout<<s.top()<<endl;
    cout<<s.size()<<endl;
    cout<<s.empty()<<endl;


    while(!s.empty()){
        cout<<s.top()<<" ";
        s.pop();
    }

    stack<int> s1,s2;
    s1.swap(s2);

    // initialize a stack
    vector<int> v = {10,20,30,40};

    stack<int , vector<int>> s3(v);

    while(!s3.empty()){
        cout<<s3.top()<<" ";
        s3.pop();
    }

    return 0;
}

File: java/stack.java
import java.util.*;

public class stack{
    public static void main(String[] args){

        // create a stack
        Stack<Integer> st = new Stack<>();

        // push
        st.push(10);
        st.push(20);
        st.push(30);

        // get the top/peek element
        System.out.println(st.peek());

        // pop the element 
        // NOTE - here pop() returns the popped element
        System.out.println(st.pop());

        // check if stack is empty or not
        System.out.println(st.isEmpty());

        // find size of stack
        System.out.println(st.size());

        // search position of an elemnt in stack -> 1 based index from top
        int idx = st.search(20);
        System.out.println(idx);
    }
}